packages/lib/src/Tools.ts

```diff
@@ -663,6 +672,24 @@ export const fromPairs = <T>(pairs: [k?: string, v?: T, ...args: unknown[]][]) =
   return r
 }
 
+/**
+ * Filters object values based on predicate
+ * @param f - Function to test values
+ * @param x - Object to filter
+ * @returns Object with only values that pass predicate
+ */
+export const filterVals = (f: (v: any) => boolean, x: Record<string, any>) => {
+  const r = {} as Record<string, any>
+
+  for (const k in x) {
+    if (f(x[k])) {
+      r[k] = x[k]
+    }
+  }
+
+  return r
+}
```

Comment:
While you're at it, see if you can add generics so we don't have to use any. `<T extends Record<string, any>>` should do the trick.